---
title: Undergraduate Thesis
description: Jeonghoon Lee's undergraduate thesis
author: Jeonghoon Lee
date: '2021-06-03'
output:
  blogdown::html_page:
    highlight: tango
slug: []
categories: []
tags: []
DisableComments: Yes
---

---

## Table of Contents

1. [Title](#title)
2. [Abstract](#abstract)
3. [Introduction](#introduction)
4. [Methodology](#methodology)
5. [Results](#results)
6. [Discussion](#discussion)
7. [Conclusion](#conclusion)
8. [Acknowledgements](#acknowledgements)
9. [References](#references)

---

# Title

Performances of functional diversity indices to randomized Poisson and binomial data distributions serving as simulations of sampling errors in species abundances

*Author: Jeonghoon Lee*

*Supervisor: Prof. Benjamin Gilbert*

*Date: 2020-04-20*

---

# Abstract

Functional diversity indices are a staple to the research of ecology and evolutionary biology. New advances are made when researchers publish innovative formulae that capture different aspects of a community’s functional diversity. This independent project investigates the performances of four different functional diversity indices when they are confronted with randomized Poisson and binomial data distributions. The four indices chosen are community weighted mean, functional dispersion, functional richness, and functional evenness. The randomized data distributions serve as simulations of field sampling errors in species abundances. This research utilized a field-collected dataset of species abundances and functional trait values in a grassland environment. True, original values of the functional diversity indices were calculated using the non-manipulated dataset. Randomizations of Poisson and binomial distributions were employed to manipulate the dataset and to replicate field sampling errors in species abundances. Errant values of the functional diversity indices were then calculated using the manipulated dataset. Simulated Poisson and binomial distributions of community-weighted mean were not different from the expected distributions. Simulated Poisson and binomial distributions of functional evenness were different from their expected distributions. The data analysis suggests that community-weighted mean is the most robust of all the chosen functional diversity indices as its simulated distributions did not deviate from its expected distributions to a significant degree. As such, future researchers should prioritize using community-weighted mean above functional evenness if they are aware of a large amount of error within their collected datasets.

---

# Introduction

Community ecology is the study of the interactions between organisms within communities on spatial and temporal scales (Sahney and Benton, 2008). Community ecologists are tasked with characterizing a community of organisms by studying their interactions with other organisms around them and their interactions with the environment. This field of ecology revolves around the designations of species richness and species diversity (Spellerberg and Fedor, 2003). Species richness measures the number of species within a community, while species diversity describes the species composition within a community. Ecologists utilize ecological indices to measure the species richness or species diversity of a community (Grabchak et al., 2017).

There are three fundamental concepts of species diversity in community ecology: alpha, beta, and gamma. These three fundamental concepts were first introduced to the field by R. H. Whittaker in 1960 (Whittaker, 1960). Alpha diversity indices focus on species within a singular community. Ecologists use alpha diversity to characterize a community on a local scale (Sepkoski, 1988). Beta diversity indices focus on comparing two different communities. Using beta diversity allows ecologists to characterize and quantify the differences in species composition between two groups of organisms (Legendre, 2007). Gamma diversity indices focus on species within a larger landscape. Gamma diversity can be used to characterize an entire region and to compare one community against many others within that specific region (Arellano and Halffter, 2003). Initially, Whittaker simply defined beta diversity as the value of the relationship between alpha (local) species diversity and gamma (regional) species diversity. Ever since their initial definitions, many researchers have strived to derive more detailed definitions of the term (Baselga and Orme, 2012). Community ecologists, in their pursuit of better methods for characterizations of communities, are on constant lookout for optimized computations of diversity indices (Chao et al., 2006).

Researchers have defined species diversity in terms of multivariate space, stating that the diversity of a community can be assessed as a function of individual, species data points of average community dissimilarity to a group centroid (Anderson et al., 2006). Furthermore, there are increasing efforts to partition species diversity into three different aspects: taxonomic, phylogenetic, and functional (Cardoso et al., 2013; Swenson, 2011). Taxonomic diversity measures the similarity or dissimilarity of communities in terms of taxa abundance and handles different taxa as being equally distinctive. Phylogenetic diversity, unlike taxonomic diversity, only takes into consideration the evolutionary relationships between different taxa. Thus, taxonomic diversity deals in distinct species, whereas phylogenetic diversity deals in families. Functional diversity measures the similarity or dissimilarity of communities in terms of functional trait abundance and disregards taxa abundance as a measure. Functional traits are the phenological characteristics of an organism (Nock et al., 2016) and can be used to quantify a particular organism. Functional diversity indices quantify the ranges and types of functional traits within communities (Mouchet et al., 2010). There are many applicable differences between functional diversity and taxonomic diversity. Taxonomic diversity comprises of log-normal distributions (Engen et al., 2008). However, functional diversity involves compounding distributions because of its reliance on both turnover and nestedness components (Bishop et al., 2015). There is also ongoing research that strives to compartmentalize functional diversity into different constituents: functional dispersion, functional richness, and functional evenness (Laliberté and Legendre, 2010; Legras et al., 2018; Villéger et al., 2008). This paper will focus on functional diversity indices primarily because studies on the robustness of taxonomic diversity indices already exist (Beck et al., 2013; Cardoso et al., 2009; Chao et al., 2006; Schroeder and Jenkins, 2018).

The functional diversity indices to be studied within this independent project are community-weighted mean (CWM), functional dispersion (FDis), functional richness (FRic), and functional evenness (FEve). CWM is defined to be the mean of a functional trait value within a community weighted by the relative abundance of a species that exhibits that functional trait (Garnier et al., 2004). CWM is calculated as:

$$CWM = \sum_{i=1}^{n} a_i x_i$$

where $a_i$ is the relative abundance of a species within community $i$ and $x_i$ is the functional trait value for that species within community $i$. FDis computes the mean distance of individual species in principal component analysis from a community-wide group centroid (Anderson, 2005). This centroid is defined by a non-Euclidean dissimilarity measure. The mean distances to the centroid, which are weighted by the relative abundances of the species, describes a multivariate dispersion. FDis is calculated as:

$$FDis = \frac{\sum_{i=1}^{n} a_i z_i}{\sum_{i=1}^{n} a_i}$$

where $a_i$ is the relative abundance of a species within community $i$ and $z_i$ is the distance of that species to a weighted centroid within community $i$. FRic and FEve both utilize the concept of an occupied niche space (Mason et al., 2005). An individual species’ niche space can be defined as the range of resources that the species can utilize within its community. FRic serves to capture the distribution of a species’ occupied niche space across the species’ range of functional trait values. FRic is calculated as:

$$FRic = \frac{SF_i}{R_i}$$

where $SF_i$ is the occupied niche space of a species within community $i$ and $R_i$ is the absolute range of the functional trait within community $i$. Meanwhile, FEve distributes the niche space across many narrow categories of species abundances. Thus, FEve studies the distribution of a range of functional trait values across the different tiers of species abundance.

Errant data is data in which there are erroneous or missing values. These inaccurate data points arise from random sampling errors. Researchers are inevitably plagued by errant data, whether they originate from non-compliance or sudden subject death. Whether they are scarce or abundant, these incorrect data points are not trivial in the least. Erroneous data points have an influence on the results that a researcher could produce. More importantly, the way that the researcher handles those erroneous points could have grave consequences on the overall analysis. Errant data can have implications on the theory of causal inference, the very backbone of the scientific process (Imai et al., 2008). To counteract errant data, more research is being done to produce statistical values that are robust (Chao et al., 2006). Such statistical values would perform well despite the presence of inaccurate data points within a dataset. Fortunately, there is mounting emphasis towards the use of such insensitive statistical measures in order to reduce the random sampling errors of estimation as much as possible. The experiment aims to answer the following questions. How can such errors alter functional diversity indices? Do random sampling errors cause systematic biases?

The primary objective of this paper is to explore the effect of randomized species abundance distributions on the performances of existing functional diversity indices. This research utilizes a real, plant community dataset to calculate the functional diversity indices. Then, the experiment will subject this real dataset to Poisson and binomial distributions. With this manipulated dataset, each functional diversity index will be calculated again for comparison to the original values. The functional diversity indices of interest are CWM, FDis, FRic, and FEve. This study aims to establish which of the four functional diversity indices are insensitive to various types of randomized distributions. By constructing such a “robustness tier”, this paper hopes to guide future ecologists in choosing the appropriate functional diversity index for their experiments.

# Methodology

All data manipulation, calculations, statistical analyses, and graphing was done in R (version 3.6.3) and RStudio (version 1.2.1335) (R Core Team, 2020).

### Preparation of the dataset

The functional traits and species abundance data were collected and generously provided by Forsyth and colleagues as part of a larger study on the effect of fragmentation on grassland plant community composition (citation pending). The plant dataset was gathered over many years at the University of Toronto’s Koffler Scientific Reserve. This long-term fragmentation experiment evaluated 14 “blocks”, which served as metacommunities. Each metacommunity was fragmented into 20 smaller communities of differing sizes: four $2m^2$ plots, eight $1m^2$ plots, and eight $0.5m^2$ plots. The research recorded four functional traits: plant height, leaf dry matter content (LDMC), leaf number, and specific leaf area (SLA). For the purposes of this experiment, focus was on the SLA and LDMC functional traits.

The functional traits and species abundance datasets were prepared for experimental simulations. The dataset was subset so that only functional trait data for $2m^2$ plots for the year 2018 remained for experimentation. Additionally, each plot was denominated to have a distinct numeric designation. For example, the second $2m^2$ plot from block 4 would be designated as “4.2”. Density distributions of SLA and LDMC values across all plots were produced in order to visualize any potential skewness in values across the plots.

### Calculation of each functional diversity index (true value)

Using the prepared datasets, the true values of community-weighted mean (CWM), functional dispersion (FDis), functional richness (FRic), and functional evenness (FEve) were calculated for each plot for the SLA functional trait. Additionally, the true values of CWM and FDis were calculated for each plot for the LDMC functional trait.

CWM was calculated using the “functcomp” function from the “FD” package (Laliberté et al., 2014). For the functional trait argument (x), a dataframe was constructed with columns for plot numbers, species names, and SLA or LDMC functional trait values. For the species abundance argument (a), a matrix was constructed with species by columns and plots by rows. All other arguments were kept at default.

FDis was calculated using the “fdisp” function from the “FD” package (Laliberté et al., 2014). For the species-by-species distance matrix argument (d), a matrix was constructed using the “gowdis” function from the “FD” package and a dataframe with columns for plot numbers, species names, and SLA or LDMC functional trait values (Laliberté et al., 2014). For the species abundance argument (a), a matrix was constructed with species by columns and plots by rows. All other arguments were kept at default.

FRic and FEve were calculated using the “REND” function from the “TPD” package (Carmona et al., 2019). The “REND” function required a trait probability density (TPD) of communities. The “TPDc” function from the “TPD” package was used to create a TPD of communities (Carmona et al., 2019). For the species abundance argument (sampUnit), a matrix was constructed with species by columns and plots by rows. For the TPD of populations argument (TPDs), an object of class “TPDsp” was created through the “TPDs” function from the “TPD” package using a vector of species names present within the dataframe (and which exceeded an abundance of one across all plots) and a vector of the species’ respective SLA functional trait values (Carmona et al., 2019). All other arguments were kept at default.

### Subjection of the dataset to randomized Poisson and binomial distributions

The species abundance dataset was manipulated to represent random sampling errors within the field. The first errant abundance treatment was the Poisson distribution, where an observation of organisms in each unit of area could act as a discrete probability model. The “rpois” function from the “stats” package was used to create randomized species abundance distributions (R Core Team, 2020). The number of datasets to be simulated was set to the number of plots. The vector giving the expected total count (ie. the lambda) was set to the original abundance vector of a species across all plots. The probability of the composition was kept at default. The second errant abundance treatment was the binomial distribution, where an observation of an organism could act as a Bernoulli trial. The “rbinom” function from the “stats” package was used to create randomized species abundance distributions (R Core Team, 2020). The number of observations to be simulated was set to the number of plots. The number of trials was set to the original abundance vector of a species across all plots. The probability was arbitrarily changed to 0.8.

For each errant abundance treatment, a thousand bootstrap simulations of the randomized abundance distributions were performed.

### Calculation of each functional diversity index (manipulated value)

For each functional diversity index, a thousand manipulated (errant) values were calculated for each plot using the original functional traits dataset and each of the thousand errant species abundance simulations produced by randomized Poisson distributions. For each functional diversity index, a mean errant value and its standard deviation were calculated for each plot. Calculations of errant functional diversity values, mean errant values, and standard deviations were repeated using errant species abundance simulations produced by randomized binomial distributions. The above calculations were conducted similarly for both SLA and LDMC functional traits.

### Comparisons between true and errant values

To visualize the distribution of mean errant values against true values, scatterplots were designed for each functional diversity index, for each errant distribution treatment, and for both SLA and LDMC functional traits.

# Results

### Functional trait density distributions

The density distributions of specific leaf area (SLA) values and leaf dry matter content (LDMC) values across all plots and for all species were heavily positive-skewed (Figures 1 and 2). The density distribution of LDMC values was more skewed than that of SLA values. There were extreme outliers within both functional trait density distributions.

### Community-weighted mean (CWM)

The distributions of mean errant CWM values had relatively low standard deviations (Figures 3, 4, 5, and 6). The distributions of mean errant CWM values were relatively even with the distributions of true CWM values (low divergence from the 1:1 line). No immediate outliers in terms of abnormal standard deviations or of divergence from the 1:1 line could be identified. However, a number of outliers with abnormally high functional index values could be identified. Distributions of mean errant CWM values calculated from randomized Poisson abundances had greater standard deviations than those calculated from randomized binomial abundances. Distributions of mean errant CWM values had lower variation and fewer outliers in terms of abnormal functional index values when the functional trait in question was SLA rather than LDMC.

### Functional dispersion (FDis)

The distributions of mean errant FDis values had relatively high standard deviations (Figures 7, 8, 9, and 10). The distributions of mean errant FDis values were relatively even with the distributions of true FDis values (low divergence from the 1:1 line). No immediate outliers in terms of abnormal standard deviations or of divergence from the 1:1 line could be identified. However, a number of outliers with abnormally low functional index values could be identified. Distributions of mean errant FDis values calculated from randomized Poisson abundances had greater standard deviations than those calculated from randomized binomial abundances. Distributions of mean errant FDis values had lower variation and fewer outliers in terms of high functional index values when the functional trait in question was SLA rather than LDMC.

### Functional richness (FRic)

The distributions of mean errant FRic values had relatively low standard deviations (Figures 11 and 12). The distributions of mean errant FRic values were relatively even with the distributions of true FRic values (low divergence from the 1:1 line). Many significant outliers in terms of abnormal standard deviations and of divergence from the 1:1 line could be identified. Additionally, many significant outliers with abnormally high functional index values could be identified. Distributions of mean errant FRic values calculated from randomized Poisson abundances had greater standard deviations than those calculated from randomized binomial abundances.

### Functional evenness (FEve)

The distributions of mean errant FEve values had relatively high standard deviations (Figures 13 and 14). The distributions of mean errant FEve values were relatively uneven with the distributions of true FEve values (moderate divergence from the 1:1 line). Many significant outliers in terms of abnormal standard deviations and of divergence from the 1:1 line could be identified. Additionally, many significant outliers with abnormally high and low functional index values could be identified. Distributions of mean errant FEve values calculated from randomized Poisson abundances had greater standard deviations than those calculated from randomized binomial abundances.

# Discussion

Community-weighted mean (CWM) appears to be the most robust functional diversity index. Distributions of mean errant CWM values have low variation, low divergence, and few outliers. CWM is calculated through the matrix multiplication of a species abundance matrix and a functional trait value matrix (Lavorel et al., 2007). The dot product nature of matrix multiplication in addition to the fundamentals of error propagation may explain the relatively low variation. The multiplicative nature of the dot product means that the random sampling errors fabricated within each cell of an errant species abundance matrix will be added in fractional quadrature. The additive nature of the dot product means that the random sampling errors now within the species’ functional trait values for a certain plot will be added in quadrature. Thus, the multiplicative nature of matrix multiplication works to lower the variation within the distributions of mean errant CWM values. Low divergence from the 1:1 line can be explained by the Central Limit Theorem. The distributions of errant CWM values for every plot are sampling distributions with the true CWM values acting as the population means for their respective sampling distributions. By increasing the number of samples (ie. the number of simulations) for a sampling distribution, that distribution will tend to a Normal distribution with the sample mean approaching the population mean. Because the number of simulations is high, it is reasonable to assume that the sample means for all plots have approached their respective population means. Outliers in functional trait values can be explained by the presence of broadleaved species in only select plots. For example, plot 1.4 contains the highest number of Bromus inermis, a rhizomatous grass. This grass species within this plot boasts the third highest SLA value out of all other species within all other plots. The relatively high contribution of a single species to a single plot coupled with the relatively high functional trait value of that species has resulted in an exceptionally high CWM value.

Functional dispersion (FDis) appears to be semi-robust. Distributions of mean errant FDis values have high variation, low divergence, and few outliers. FDis is calculated through the “mean distance in multidimensional functional trait space of individual species to the centroid of all species” (Laliberté and Legendre, 2010). The dual-summative nature of FDis calculation and the error propagation fundamentals may explain the relatively high variation. The dual-summative nature of the FDis calculation means that the random sampling errors fabricated within each cell of an errant species abundance matrix will be added in quadrature twice. The multiplicative nature of the FDis calculation means that the random sampling errors now within each species distance from a group centroid will be added in fractional quadrature. Thus, the dual-summative nature of FDis calculation works to raise the variation within the distributions of mean errant FDis values. One can now notice that the multiplicative and additive phases of CWM and FDis calculations are reversed. For CWM, multiplication precedes addition to propagate the random sampling errors within species abundances, resulting in low variation. For FDis, addition precedes multiplication to propagate the random sampling errors, resulting in high variation. This suggests that the order of addition-multiplication within an index calculation affects the variation within the index distribution. Low divergence from the 1:1 line can be explained by the Central Limit Theorem as previously described for CWM. Outliers in functional trait values can be explained by the abundance of narrow-leaved species. For example, plot 9.2 predominantly contains a high number of Festuca rubra, a short-ligule grass. This grass species within this plot possesses the second lowest SLA value out of all other species. The relatively low distance of the prevalent species within a single plot to a high-value centroid has resulted in an exceptionally low FDis value.

Functional richness (FRic) appears to be semi-robust. Distributions of mean errant FRic values have low variation, low divergence, and a number of significant outliers. FRic is calculated through the division of a species’ niche space within a community by the absolute range of the functional trait (Mason et al., 2005). Once again, the multiplicative nature of FRic calculation and the error propagation fundamentals can explain the relatively low variation. The multiplicative nature of the FRic calculation means that the random sampling errors within each species’ niche space will be added in quadrature. Thus, the multiplicative nature of FRic calculation works to lower the variation within the distributions of mean errant FRic values. Low divergence from the 1:1 line can be explained by the Central Limit Theorem as previously described for CWM. Significant outliers in variation, divergence, and functional trait values can be explained by the skewness of functional trait distributions. It can be proposed that the skewed trait distributions have created inflated niche spaces for certain communities and for certain species. These niche spaces also contain high variation because of the outliers within and the spread of the trait distributions. The highly variant and inflated niche spaces could have produced an exceptionally high FRic value.

Functional evenness (FEve) appears to be the least robust functional diversity index. Distributions of mean errant FEve values have high variation, moderate divergence, and multiple significant outliers. The potentially infinite multiplicative nature of FEve calculation can explain the relatively high variation within the mean errant FEve distributions. Because FEve calculation involves dividing a species’ occupied niche space into many narrow categories, with each category being multiplied by a functional trait’s evenness index, there are numerous multiplicative phases to the FEve calculation. Even though the random sampling errors within each species abundance category is only added in quadrature, the potentially infinite characteristic of dividing the occupied niche space into greater and greater numbers of categories works to raise the index variation. Moderate divergence from the 1:1 line can be explained by the Central Limit Theorem as previously described for CWM as well as the presence of significant outliers. Significant outliers in variation, divergence, and functional trait values can be explained by the skewness of functional trait distributions. As before, it is reasonable to assume that the skewed trait distributions have created inflated niche spaces for certain communities and for certain species. These niche spaces also contain high variation because of the outliers within and the spread of the trait distributions. The highly variant and inflated niche spaces could have produced an exceptionally high FEve value. The division of other, deflated niche spaces into many narrow categories could have produced an exceptionally low FEve value.

Furthermore, all mean errant functional diversity distributions have systematically less variation when the species abundances were fabricated by randomized binomial distributions. In other words, Poisson variance is greater than binomial variance (λ > np(1 – p)). This is because the lambda (λ) argument within the “rpois” function was equal to the size (n) argument within the “rbinom” function. However, binomial variance is mathematically smaller than Poisson variance because of the inclusion of the p(1 – p) expression within its calculation and the probability (p) argument. Thus, binomial variance will always be smaller than Poisson variance. In this experiment, probability is set to 0.8, so binomial variance is smaller by a factor of 0.16 (p(1 – p) = 0.16).

The positive-skewness of the functional trait distributions may have predisposed the functional diversity indices to perform poorly. It is noticeable how index distributions suffered from higher variation when the functional trait was LDMC rather than SLA. The LDMC density distribution is more positive-skewed than that of SLA. Skewness in trait distributions leads to higher variation within index distributions, insinuating that skewness has a detrimental impact on the performances of functional diversity indices. Additionally, many significant outliers within the mean errant index distributions can be attributed to the skewed functional trait distributions. Nevertheless, some diversity indices such as CWM still performed admirably in the face of skewed trait distributions. Despite the pronounced skewness, distributions of mean errant CWM values had relatively low variations. This shows that functional diversity indices such as CWM are relatively robust to skewed functional trait distributions.

# Conclusion

Unfortunately, field research errors are inevitable. Prudent community ecologists do well to take preventive measures that would minimize the presence of field errors and to select analytical methods that confer accurate scientific reporting. Through experimentation with randomized species abundances, it can be concluded that community-weighted mean (CWM) is the most robust functional diversity index. CWM calculation (ie. matrix multiplication) seems to be resistant to the propagation of random sampling errors. Functional dispersion (FDis) is also relatively robust. On the other hand, functional evenness (FEve) seems to be the most sensitive functional diversity index. The calculation and use of a species’ niche space within a community leaves the index highly vulnerable to random sampling errors and dependent on the distribution of the functional traits in question. Community ecologists must employ FEve and, to a lesser extent, functional richness (FRic) with caution.

There are a number of aspects within this experiment that could use improvement and extension. Further research could be done to investigate how changing the probability weight of the binomial distribution can impact the performances of functional diversity indices. In scenarios where the probability weight is close to 0 or close to 1, the variation within the functional diversity index distributions may decrease even further. In scenarios where the probability weight is close to 0.5, the variation within the index distributions may increase as the binomial variance has reached its maximum value. One could also examine the performances of functional diversity indices when species abundances are fabricated by randomized negative binomial distributions. A negative binomial distribution outlines the number of failures in Bernoulli trials to achieve a fixed number of successes. This can be translated to the number of absences before a certain number of plant species has been observed. Comparisons between distribution variances will be difficult as the negative binomial variance (rp/(1 – p)2, where r is the number of failures and p is the probability of observation) is not easily analogous to that of Poisson or binomial distributions. Finally, more study must be done on the performances of FRic and FEve and on the importance of the absolute range of the functional trait. During the calculations of FRic and FEve, the integrals of the probability density functions of certain communities did not equal one and had to be rescaled. To solve this issue, the ranges of the functional trait could be increased. However, this raises the question of how sensitive the indices are to the definitions of the trait ranges.

Researchers may be tempted to use a certain functional diversity index because it comes with the promise of less “loss of information”. Just because an index tells a finer story, it does not mean that they tell that story well. The interpretations within this manuscript will hopefully serve as a tool with which to judge the use of CWM, FDis, FRic, and FEve within future community ecology experiments.

# Acknowledgements

Special thanks to Leila Forsyth for providing the dataset and providing support throughout the project. Special thanks to Koffler Scientific Reserve at Joker’s Hill for allowing collection of data on its grassland plant species.

# References

Anderson, M. J. 2005. Distance-Based Tests for Homogeneity of Multivariate Dispersions. Biometrics 62:245–253.

Anderson, M. J., K. E. Ellingsen, and B. H. Mcardle. 2006. Multivariate dispersion as a measure of beta diversity. Ecology Letters 9:683–693.

Arellano, L., and G. Halffter. 2003. Gamma diversity: derived from and a determinant of Alpha diversity and Beta diversity. An analysis of three tropical landscapes. Acta zoológica mexicana, 90:27–76.

Baselga, A., and C. D. L. Orme. 2012. betapart: an R package for the study of beta diversity. Methods in Ecology and Evolution 3:808–812.

Beck, J., J. D. Holloway, and W. Schwanghart. 2013. Undersampling and the measurement of beta diversity. Methods in Ecology and Evolution 4:370–382.

Bishop, T. R., M. P. Robertson, B. J. V. Rensburg, and C. L. Parr. 2015. Contrasting species and functional beta diversity in montane ant assemblages. Journal of Biogeography 42:1776–1786.

Cardoso, P., F. Rigal, J. C. Carvalho, M. Fortelius, P. A. V. Borges, J. Podani, and D. Schmera. 2013. Partitioning taxon, phylogenetic and functional beta diversity into replacement and richness difference components. Journal of Biogeography 41:749–761.

Cardoso, P., P. A. V. Borges, and J. A. Veech. 2009. Testing the performance of beta diversity measures based on incidence data: the robustness to undersampling. Diversity and Distributions 15:1081–1090.

Carmona, C. P., F. Bello, N. W. H. Mason, and J. Lepš. 2019. Trait probability density (TPD): measuring functional diversity across scales based on TPD with R. Ecology 100.

Chao, A., R. L. Chazdon, R. K. Colwell, and T.-J. Shen. 2006. Abundance-Based Similarity Indices and Their Estimation When There Are Unseen Species in Samples. Biometrics 62:361–371.

Engen, S., B.-E. Sæther, A. Sverdrup-Thygeson, V. Grøtan, and F. Ødegaard. 2008. Assessment of species diversity from species abundance distributions at different localities. Oikos 117:738–748.

Garnier, E., J. Cortez, G. Billès, M.-L. Navas, C. Roumet, M. Debussche, G. Laurent, A. Blanchard, D. Aubry, A. Bellmann, C. Neill, and J.-P. Toussaint. 2004. Plant Functional Markers Capture Ecosystem Properties During Secondary Succession. Ecology 85:2630–2637.

Grabchak, M., E. Marcon, G. Lang, and Z. Zhang. 2017. The generalized Simpson’s entropy is a measure of biodiversity. Plos One 12.

Imai, K., G. King, and E. A. Stuart. 2008. Misunderstandings between experimentalists and observationalists about causal inference. Journal of the Royal Statistical Society: Series A (Statistics in Society) 171:481–502.

Laliberté, E., and P. Legendre. 2010. A distance-based framework for measuring functional diversity from multiple traits. Ecology 91:299–305.

Laliberté, E., P. Legendre, and B. Shipley. 2014. FD: measuring functional diversity from multiple traits, and other tools for functional ecology. R package version 1.0-12.

Lavorel, S., K. Grigulis, S. Mcintyre, N. S. G. Williams, D. Garden, J. Dorrough, S. Berman, F. Quétier, A. Thébault, and A. Bonis. 2007. Assessing functional diversity in the field – methodology matters! Functional Ecology 0.

Legendre, P. 2007. Studying beta diversity: ecological variation partitioning by multiple regression and canonical analysis. Journal of Plant Ecology 1:3–8.

Legras, G., N. Loiseau, and J.-C. Gaertner. 2018. Functional richness: Overview of indices and underlying concepts. Acta Oecologica 87:34–44.

Mason, N. W. H., D. Mouillot, W. G. Lee, and J. B. Wilson. 2005. Functional richness, functional evenness and functional divergence: the primary components of functional diversity. Oikos 111:112–118.

Mouchet, M. A., S. Villéger, N. W. H. Mason, and D. Mouillot. 2010. Functional diversity measures: an overview of their redundancy and their ability to discriminate community assembly rules. Functional Ecology 24:867–876.

Nock, C. A., R. J. Vogt, and B. E. Beisner. 2016. Functional Traits. eLS:1–8.

R Core Team. 2020. R: A language and environment for statistical computing. R Foundation for Statistical Computing. https://www.R-project.org/.

Sahney, S., and M. J. Benton. 2008. Recovery from the most profound mass extinction of all time. Proceedings of the Royal Society B: Biological Sciences 275:759–765.

Schroeder, P. J., and D. G. Jenkins. 2018. How robust are popular beta diversity indices to sampling error? Ecosphere 9.

Sepkoski, J. J. 1988. Alpha, beta, or gamma: where does all the diversity go? Paleobiology 14:221–234.

Spellerberg, I. F., and P. J. Fedor. 2003. A tribute to Claude Shannon (1916-2001) and a plea for more rigorous use of species richness, species diversity and the ‘Shannon-Wiener’ Index. Global Ecology and Biogeography 12:177–179.

Swenson, N. G. 2011. Phylogenetic Beta Diversity Metrics, Trait Evolution and Inferring the Functional Beta Diversity of Communities. PLoS ONE 6.

Villéger, S., N. W. H. Mason, and D. Mouillot. 2008. New Multidimensional Functional Diversity Indices For A Multifaceted Framework In Functional Ecology. Ecology 89:2290–2301.

Whittaker, R. H. 1960. Vegetation of the Siskiyou Mountains, Oregon and California. Ecological Monographs 30:407–407.